src/components/data/utils.js

```diff
@@ -21,14 +21,23 @@ function getEncodedPath(path) {
 /**
  *  Get storage id from router path
  *
+ *  Sample path: /data/ds/iplant/home/ipcdev
+ *
  * @param {string} pathname - path from the router
  * @returns {string|*} storage id
  */
 function getStorageIdFromPath(pathname) {
     if (!pathname) {
         return "";
     }
-    return pathname.split(constants.PATH_SEPARATOR)[2];
+
+    const paths = pathname.split(constants.PATH_SEPARATOR);
+    if (paths.length > 2) {
+        return paths[2];
+    } else {
+        // Default to data store.  Needed if user navigates to just /data.
+        return "ds";
```

Comment:
If user lands on `/data`, the corresponding `index.js` file will forward the request to `listing.js` component with empty path which in turn will forward the request to `/data/ds/<user home path>. 

`useEffect(() => {
        //route to default path
        if (dataRoots.length > 0 && !path) {
            handlePathChange(dataRoots[0].path);
        }
    }, [dataRoots, handlePathChange, path]);`
 

So router path will already have the storage id.

Anyway I do think it is a good idea to have `ds` as default storage id. But just want to make sure we are all on the same page when it comes to how the code works.

Reply:
I made these changes super late on a Friday so there are probably bugs 😄 Thanks for reviewing

When I navigated to `/data` without this change, I ended going to `/data/undefined/iplant/home/aramsey` I believe.  I'll double all the handler calls to see what's needed and what's not.

And unfortunately no, this was not intended to fix the reloading, just intended to get the data view to be reusable.  I happened to notice the reloading while I was making these changes so I made the comment in the UI channel.  If needed, I can look into the reloading issue after this PR.

Reply:
OK, I just ran through the code again. The default in utils is needed because if the user goes to `/data` there is no storage ID.  

The Listing fetches the roots, then once the roots come back, Listing calls `handlePathChange` with the root home path "/iplant/home/userId". Then the page tries to get the storage ID using `router.pathname` which is still just `/data` at that point.  So we end up with `/data/undefined/iplant/home/userId`. Adding the default `ds` in the `utils` file though, updates the path to `/data/ds/iplant/home/userId`, and from then on `router.pathname` will find a storage ID.

You might be thinking of this bit of code that used to be in the Listing, but has been removed to get routing out of our components.  This would default to adding `ds` in the path: https://github.com/cyverse-de/sonora/blob/bf4649bd8930437be0b337bb462a9f845553e21f/src/components/data/listing/Listing.js#L250-L256

Reply:
Oh right I was purely commenting  based on how I had coded. Sorry about the confusion.